### ISSUE TITLE
How to return GZIP Content-Encoding

### ISSUE THREAD

**Issue by fpozzobon** [slank/awsgi#26]
Hi,

I am trying to return a GZIP encoded response but getting the error :

``` 
{
    "errorMessage": "'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte",
    "errorType": "UnicodeDecodeError",
    "stackTrace": [
        "  File \"/var/task/lambda_handler.py\", line 75, in handler\n    return awsgi.response(app, event, context)\n",
        "  File \"/var/task/awsgi/__init__.py\", line 22, in response\n    return sr.response(output)\n",
        "  File \"/var/task/awsgi/__init__.py\", line 41, in response\n    'body': self.body.getvalue() + ''.join(map(convert_str, output)),\n",
        "  File \"/var/task/awsgi/__init__.py\", line 9, in convert_str\n    return s.decode('utf-8') if isinstance(s, bytes) else s\n"
    ]
}

```
Is there any way to skip the response transformed in String ?

**Comment by slank** [comment]
@fpozzobon Does [this](https://stackoverflow.com/questions/39453097/how-to-return-gzipped-content-with-aws-api-gateway) StackOverflow question help?

I've never needed to return content through this module in other encodings. I'm happy to help, but I think I'll need more information. Could you make a small complete code example that produces the error? I think it's important to see how you're calling the APIs and what data you're passing in headers and body.

**Comment by fpozzobon** [comment]
Hi @slank thanks for your quick reply :)
Please find below how we zip the content, I am using the following annotation on top of the route : 
``` 
def gzipped(f):
    @functools.wraps(f)
    def view_func(*args, **kwargs):

        @after_this_request
        def zipper(response):
            accept_encoding = request.headers.get('Accept-Encoding', '')

            if 'gzip' not in accept_encoding.lower():
                return response

            response.direct_passthrough = False

            if (response.status_code < 200 or
                response.status_code >= 300 or
                'Content-Encoding' in response.headers):
                return response
            gzip_buffer = IO()
            gzip_file = gzip.GzipFile(mode='wb',
                                      fileobj=gzip_buffer)
            gzip_file.write(response.data)
            gzip_file.close()

            response.data = gzip_buffer.getvalue()
            response.headers['Content-Encoding'] = 'gzip'
            response.headers['Vary'] = 'Accept-Encoding'
            response.headers['Content-Length'] = len(response.data)

            return response

        return f(*args, **kwargs)

    return view_func
```

**Comment by cwalv** [comment]
This isn't possible with `isBase64Encoded` [hard-coded to False](https://github.com/slank/awsgi/blob/master/awsgi/__init__.py#L38). There's [a fork](https://github.com/truckpad/aws-lambda-wsgi/blob/master/aws_lambda_wsgi/__init__.py#L43) of this project that checks for common binary `Content-Type` headers and then encodes based on that.

I don't think this will really do what you want though (i.e., send a gzipped response to the client), because lambda's have to be exposed with an [ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html) or API Gateway; the overhead of gzipping and then base64 encoding will only slow your function down.

**Comment by slank** [comment]
This is fixed in v0.1.0
